RHIDP-1925 Fix GitHub integration procedure (#646)
redhat-developer/red-hat-developers-documentation-rhdh: modules/authentication/proc-enabling-authentication-with-github.adoc

        enterpriseInstanceUrl: ${GITHUB_HOST_DOMAIN}
----

[TIP]
====

> To enable GitHub integration with another authentication provider you have already configured, omit the `auth` and `signInPage` sections from this procedure:

Author: ```suggestion
To enable GitHub integration with another authentication provider you have already configured:

* Add the GitHub provider to the existing `auth` section.
* Keep the `signInPage` section from your authentication provider.
```

> To enable GitHub integration with another authentication provider you have already configured, omit the `auth` and `signInPage` sections from this procedure:

Reviewer: @themr0c adding the GitHub provider is not necessary, but if it's already there, there's no need to omit/remove it.

        enterpriseInstanceUrl: ${GITHUB_HOST_DOMAIN}
----

[TIP]
====
To enable GitHub integration with another authentication provider you have already configured:

* Add the GitHub provider to the existing `auth` section.
* Keep the `signInPage` section from your authentication provider.

.`app-config-rhdh.yaml` fragment with mandatory fields to enable integration with GitHub and use a different authentication provider
[source,yaml]
----

> integrations:
>   github:
>     - host: ${GITHUB_HOST_DOMAIN}
>       apps:
>         - appId: ${AUTH_GITHUB_APP_ID}
>           clientId: ${AUTH_GITHUB_CLIENT_ID}
>           clientSecret: ${GITHUB_CLIENT_SECRET}
>           webhookUrl: ${GITHUB_WEBHOOK_URL}
>           webhookSecret: ${GITHUB_WEBHOOK_SECRET}
>           privateKey: |
>             ${GITHUB_PRIVATE_KEY_FILE}

Author: ```suggestion
auth:
  environment: production
  providers:
    github:
      production:
        clientId: ${AUTH_GITHUB_CLIENT_ID}
        clientSecret: ${AUTH_GITHUB_CLIENT_SECRET}
    __<your_other_authentication_providers_configuration>__
integrations:
  github:
    - host: ${GITHUB_HOST_DOMAIN}
      apps:
        - appId: ${AUTH_GITHUB_APP_ID}
          clientId: ${AUTH_GITHUB_CLIENT_ID}
          clientSecret: ${GITHUB_CLIENT_SECRET}
          webhookUrl: ${GITHUB_WEBHOOK_URL}
          webhookSecret: ${GITHUB_WEBHOOK_SECRET}
          privateKey: |
            ${GITHUB_PRIVATE_KEY_FILE}
signInPage: __<your_main_authentication_provider>__
```

        enterpriseInstanceUrl: ${GITHUB_HOST_DOMAIN}
----

[TIP]
====

> To enable GitHub integration with another authentication provider you have already configured:

Reviewer 2: ```suggestion
To enable GitHub integration with a different authentication provider, complete the following configurations:
```

* Add the GitHub provider to the existing `auth` section.

> * Keep the `signInPage` section from your authentication provider.

Reviewer 2: ```suggestion
* Keep the `signInPage` section from your authentication provider configuration.
```

> .`app-config-rhdh.yaml` fragment with mandatory fields to enable integration with GitHub and use a different authentication provider

Reviewer 2: ```suggestion
.`app-config-rhdh.yaml` fragment with mandatory fields to enable GitHub integration and use a different authentication provider
```

> [source,yaml]

Author: ```suggestion
[source,yaml,subs="+quotes"]
```
